src/pages/Profile.js

```diff
@@ -1,37 +1,268 @@
+        profile,
+        parent: parentProfile
+      });
+    })();
+  }, [client, geoId]);
+
+  useEffect(() => {
+    if (!profiles.isLoading) {
+      (async () => {
+        setChartsData({
+          isLoading: true
+        });
+
+        const parent = {
+          geoLevel: profiles.parent.parentLevel,
+          geoCode: profiles.parent.parentCode
+        };
+        const { data: profileVisualsData } = await client.query({
+          query: buildVisualsQuery(visuals, parent),
+          variables: {
+            geoCode: profiles.profile.geoCode,
+            geoLevel: profiles.profile.geoLevel
+          }
+        });
+
+        setChartsData({
+          isLoading: false,
+          profileVisualsData
+        });
+      })();
+    }
+  }, [profiles, visuals, client]);
+
+  // get all available profiletabs
+  const profileTabs = useMemo(
+    () => [
+      {
+        name: 'All',
+        slug: 'all'
+      },
+      ...sectionedCharts
+        .map((section, i) => ({
+          ...section,
+          index: i
+        }))
+        // Filter empty sections
+        .filter(
+          section =>
+            section.charts.filter(
+              chart =>
+                chartData.isLoading ||
+                !chart.visuals.find(
+                  visual =>
+                    !chartData.profileVisualsData ||
+                    chartData.profileVisualsData[visual.id].nodes.length === 0
+                )
+            ).length !== 0
+        )
+        .map(section => ({
+          name: section.sectionTitle,
+          slug: slugify(section.sectionTitle),
+          index: section.index
+        }))
+    ],
+    [chartData]
+  );
+
+  /**
+   * Victory renders take alot of time
+   * causing a few seconds UI block which is bad UX.
+   * This caches the components so they do not have to render again.
+   */
+  const chartComponents = useMemo(
+    () =>
+      profileTabs.slice(1).map(tab => (
+        <Grid item container spacing={4} id={tab.slug} key={tab.slug}>
+          {/* <ProfileSectionTitle loading={chartData.isLoading} tab={tab} /> */}
+          {sectionedCharts[tab.index].charts
+            .filter(
+              ({ visuals: v }) =>
+                chartData.isLoading ||
+                (chartData.profileVisualsData &&
+                  /* data is not missing */
+                  !v.find(
+                    ({ id }) =>
+                      chartData.profileVisualsData[id].nodes.length === 0
+                  ))
+            )
+            .map(chart => (
+              <div style={{ margin: '40px 0', maxWidth: '100%' }}>
+                <InsightContainer
+                  classes={{ root: classes.chart }}
+                  key={chart.id}
+                  loading={chartData.isLoading}
+                  title={chart.title}
+                  source={{
+                    title: 'Community Survey 2016',
+                    href: 'http://dev.dominion.africa'
+                  }}
+                >
```

Comment:
Lets add insight links for the **Read Country Analysis** button... I think it's just `/profiles/country_slug`

Reply:
Lets do this in a separate PR when we are sure about it.
This is just initial setup.